extensions/sources/sql/ext/sql.go

```diff
@@ -89,6 +96,16 @@ func (m *sqlsource) Open(ctx api.StreamContext, consumer chan<- api.SourceTuple,
 			logger.Debugf("Query the database with %s", query)
 			rows, err := m.db.Query(query)
 			if err != nil {
+				logger.Errorf("sql source meet error, err:%v", err)
+				canReconnect, err2 := m.Reconnect(err)
+				if canReconnect {
+					if err2 == nil {
+						continue
+					} else {
+						logger.Errorf("Run sql query(%s) error %v", query, err2)
+						errCh <- err2
```

Comment:
Do not send out recoverable error to error channel, it will stop the whole rule. Just send out an ErrorSourceTuple